var/spack/repos/builtin/packages/gaudi/package.py

```diff
@@ -45,6 +45,8 @@ class Gaudi(CMakePackage):
     # fixes for the cmake config which could not find newer boost versions
     patch("link_target_fixes.patch", when="@33.0:34")
     patch("link_target_fixes32.patch", when="@:32.2")
+    # remove use std::ofstream instead of deprecated boost::filesystem
+    patch("rm-boost-ofstream.patch", when="@34:")
```

Comment:
Upstream patch now available and merged, so potentially this is cleaner with
```suggestion
    patch('https://gitlab.cern.ch/gaudi/Gaudi/-/commit/03db151003d5207795ae8ec96aaf2f5f8a9ea008.diff',
          sha256='e73f45f10ea4a143a05c53a468233ceaad1a8ec22dee9877d178a3c9ee46c70f',
          when='@:36.5')
```
and then remove the local copy of the patch.